server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -219,9 +219,6 @@ public IndexAbstraction resolveWriteIndexAbstraction(ClusterState state, DocWrit
 
         if (expressions.size() == 1) {
             IndexAbstraction ia = state.metadata().getIndicesLookup().get(expressions.iterator().next());
-            if (ia == null) {
-                throw new IndexNotFoundException(expressions.iterator().next());
-            }
```

Comment:
This check that is removed is made redundant by the `resolveExpressions` invocation on line 218 above.